util/merkle-mountain-range/src/helper.rs

```diff
@@ -0,0 +1,110 @@
+pub fn leaf_index_to_pos(index: u64) -> u64 {
+    if index == 0 {
+        return 0;
+    }
+    // leaf_count
+    let mut leaves = index + 1;
+    let mut tree_node_count = 0;
+    let mut height = 0u32;
+    while leaves > 1 {
+        // get heighest peak height
+        height = (leaves as f64).log2() as u32;
```

Comment:
A simple way of log2 is 
```
fn log_2(x: u32) -> u32 {
    assert!(x > 0);
    32 - x.leading_zeros() - 1
}

assert_eq!(log_2(8), 3);
assert_eq!(log_2(10), 3);
```

Reply:
Not sure it can work in all scenarios. See the benchmark, `log2` is not a hot spot yet.